adsft/tests/test_extraction.py

```diff
@@ -345,73 +286,32 @@ def test_that_table_is_extracted_correctly(self):
 
         full_text_content = self.extractor.open_xml()
 
-        if self.parsers:
-            for parser_name in self.parsers:
-
-                s = u"TABLE I. TEXT a NOTES a TEXT"
-
-                # we know that html5lib does not extract tables correctly in all cases
-                if parser_name == "html5lib":
-                    s = u"TABLE I. TEXT a"
-
-                self.extractor.parse_xml(preferred_parser_names=(parser_name,))
-                section = self.extractor.extract_string('//body//table-wrap')
-                self.assertEqual(section, s)
-        else:
-            self.extractor.parse_xml()
+        s = u"TABLE I. TEXT a NOTES a TEXT"
+        for parser_name in self.preferred_parser_names:
+            self.extractor.parse_xml(preferred_parser_names=(parser_name,))
             section = self.extractor.extract_string('//body//table-wrap')
             self.assertEqual(section, s)
 
-    def test_handling_of_parsers_that_remove_body_tag(self):
+    def test_body_tag(self):
```

Comment:
I don't think merging test_handling_of_parsers_that_remove_body_tag and test_handling_of_parsers_that_detect_namespaces to abide by DRY is the best way to go because it is not clear which parser is failing and therefore which of these cases is occurring, but the unit tests need to undergo a refactoring in general so I think this can be addressed in a separate PR.